src/Compilers/CSharp/Portable/Symbols/Extensions/SynthesizedExtensionMarker.cs

```diff
@@ -78,6 +78,29 @@ protected override (TypeWithAnnotations ReturnType, ImmutableArray<ParameterSymb
                 if (parameter is { })
                 {
                     checkUnderspecifiedGenericExtension(parameter, ContainingType.TypeParameters, diagnostics);
+
+                    TypeSymbol parameterType = parameter.TypeWithAnnotations.Type;
+                    RefKind parameterRefKind = parameter.RefKind;
+                    SyntaxNode? parameterTypeSyntax = parameterList.Parameters[0].Type;
+                    Debug.Assert(parameterTypeSyntax is not null);
+
+                    if (!parameterType.IsValidExtensionParameterType())
+                    {
+                        diagnostics.Add(ErrorCode.ERR_BadTypeforThis, parameterTypeSyntax, parameterType);
+                    }
+                    else if (parameterRefKind == RefKind.Ref && !parameterType.IsValueType)
+                    {
+                        diagnostics.Add(ErrorCode.ERR_RefExtensionParameterMustBeValueTypeOrConstrainedToOne, parameterTypeSyntax);
+                    }
+                    else if (parameterRefKind is RefKind.In or RefKind.RefReadOnlyParameter && parameterType.TypeKind != TypeKind.Struct)
+                    {
+                        diagnostics.Add(ErrorCode.ERR_InExtensionParameterMustBeValueType, parameterTypeSyntax);
+                    }
+
+                    if (parameter.Name is "" && parameterRefKind != RefKind.None)
```

Comment:
>parameterRefKind != RefKind.None [](http://example.com/codeflow?start=48&length=32)

Should `scoped` be disallowed as well?

Reply:
Yes, but that's currently a parsing error. See `ReceiverParameterValidation_UnnamedReceiverParameter_Scoped`, which has a comment for follow-up